ngraph_bridge/ngraph_encapsulate_op.cc

```diff
@@ -53,1019 +57,638 @@ namespace ng = ngraph;
 #endif
 
-    for (size_t i = 0; i < input_shapes.size(); i++) {
+  int time_create_or_lookup_tensors = create_or_lookup_tensors.ElapsedInMS();
+
+  // Execute the nGraph function.
+  ngraph::Event event_execute_function("Execute nGraph", name(), "");
+  Timer execute_function;
+  {
+    BackendManager::LockBackend(ng_encap_impl.GetOpBackend());
+    NGRAPH_VLOG(4) << "NGraphEncapsulateOp::Compute call starting for cluster "
+                   << ng_encap_impl.GetNgraphCluster();
+    try {
+      ng_exec->call(ng_outputs, ng_inputs);
+    } catch (const std::exception& exp) {
+      ng_function = ng_encap_impl.GetNgFunctionMap()[ng_exec];
+      BackendManager::UnlockBackend(ng_encap_impl.GetOpBackend());
+      NgraphSerialize("tf_function_error_" + ctx->op_kernel().name() + ".json",
+                      ng_function);
+      OP_REQUIRES(ctx, false,
+                  errors::Internal(
+                      "Caught exception while executing nGraph computation: ",
+                      exp.what(), "\n"));
+    } catch (...) {
+      ng_function = ng_encap_impl.GetNgFunctionMap()[ng_exec];
+      BackendManager::UnlockBackend(ng_encap_impl.GetOpBackend());
+      NgraphSerialize("tf_function_error_" + ctx->op_kernel().name() + ".json",
+                      ng_function);
+      OP_REQUIRES(
+          ctx, false,
+          errors::Internal("Error in executing the nGraph computation\n"));
+    }
+    BackendManager::UnlockBackend(ng_encap_impl.GetOpBackend());
+  }
+  int time_execute_function = execute_function.ElapsedInMS();
+  event_execute_function.Stop();
+
+  long vm, rss;
+  MemoryProfile(vm, rss);
+  NGRAPH_VLOG(1) << "NGRAPH_TF_MEM_PROFILE:  OP_ID: "
+                 << ng_encap_impl.GetInstanceId() << " Step_ID: " << step_id
+                 << " Cluster: " << name() << " Input Tensors created: "
+                 << ng_input_tensor_size_in_bytes / (1024 * 1024) << " MB"
+                 << " Output Tensors created: "
+                 << ng_output_tensor_size_in_bytes / (1024 * 1024) << " MB"
+                 << " Total process memory: " << rss / (1024 * 1024) << " GB";
+
+  NGRAPH_VLOG(4) << "NGraphEncapsulateOp::Compute call done for cluster "
+                 << ng_encap_impl.GetNgraphCluster();
+
+  // Copy value to host if backend is not CPU
+  ngraph::Event event_copy_output("Output - copy back", name(), "");
+  Timer copy_output_tensors_to_host;
+
+  try {
+    size_t output_tensor_count = output_caches.size();
+    std::vector<std::unique_ptr<ngraph::Event>> output_copy_events;
 #if defined(NGRAPH_TF_ENABLE_VARIABLES_AND_OPTIMIZERS)
-      bool ref_exists = NGraphCatalog::ExistsInInputVariableSharedNameMap(
-          m_graph_id, def().name(), i);
+    if (ng_encap_impl.GetNumberOfOutputs() == -1) {
+      NGRAPH_VLOG(4) << "Settig number of outputs for " << def().name();
+      ng_encap_impl.SetNumberOfOutputs(output_caches.size());
+    }
+    if (ng_encap_impl.GetNumberOfOutputs() == -1) {
```

Comment:
Only 1 if block is required, that sets both inputs and outputs

Reply:
looks like a merge master error, will fix it